nx/lib/nx.ex

```diff
@@ -16679,6 +16679,74 @@ defmodule Nx do
     |> as_type(opts[:type])
   end
 
+  @doc """
+  Creates an Nx-tensor from an already-allocated memory space.
+
+  This function should be used with caution, as it is very backend-specific.
+
+  The `backend` argument is the backend module (such as `Nx.BinaryBackend`).
+  `opaque_pointer` is the corresponding value that would be returned from
+  a call to `get_pointer/2`.
+
+  ## Options
+
+    * `:names` - refer to `tensor/2`
+    * `:backend_opts` - options specific to the backend for handling
+      the pointer.
+
+  ## Examples
+
+      Nx.from_pointer(<<1, 2, 3, 4>>, {:s, 64}, {1, 3})
+      #Nx.Tensor<
+        s64[1][3]
+        [
+          [10, 20, 30]
+        ]
+      >
+
+      Nx.from_pointer(<<5, 6, 7>>, {:s, 64}, {1, 3}, names: [nil, :col], backend_opts: [some: :option, another: :option])
+      #Nx.Tensor<
+        s64[1][col: 3]
+        [
+          [10, 20, 30]
+        ]
+      >
+  """
+  @doc type: :creation
+  def from_pointer(backend, opaque_pointer, type, shape, opts \\ []) when is_atom(backend) do
```

Comment:
Sorry, I gave bad advice. The backend should be an option, similar to how it is done in `from_binary`. :) So there is a single option called `:backend`.

Regarding the backend options, we could simplify proxy everything else to the backend. Or we should call it `pointer_options` to avoid confusion.

Reply:
I don't like backend being an option here because it is mandatory and there's no sensible default we can have.

Reply:
I agree that pointer_options is a better name! Will change both in the next commit

Reply:
If you want to force the backend, then it should mirror the API from backend_transfer, where the backend is either an atom or a two-element tuple. Also note the examples above are incorrect (they are not passing the backend as first argument).